src/pages/Profile.jsx

```diff
@@ -9,16 +9,64 @@ import {
 import { useLocation, useNavigate } from "react-router-dom";
 
 import { FaUserCircle } from "react-icons/fa";
+import { useState } from "react";
+import {auth,storage,} from "../lib/firebase";
+import { useSnackbar } from "notistack";
 
 function Profile() {
   const { name, email, avatar } = useLocation().state;
   const isNonMobile = useMediaQuery("(min-width: 768px)");
+  const { enqueueSnackbar } = useSnackbar();
+  const [image, setImage] = useState("");
+  const [img, setAvatar] = useState(avatar);
+  const [visible,setVisibility] = useState(false);
   const navigate = useNavigate();
 
   const handleBack = () => {
     navigate("/dummygram"); // Use navigate function to change the URL
   };
 
+  const handleChange = (e) => {
+    if (e.target.files[0]) {
+      setAvatar(URL.createObjectURL(e.target.files[0]));
+      setImage(e.target.files[0]);
+      setVisibility(true);
+    }
+  };
+  console.log(name)
+  console.log(auth.currentUser.displayName)
+  const handleSave = async() => {
+    const uploadTask =  storage.ref(`images/${image?.name}`).put(image);
+   await uploadTask.on(
+      "state_changed",
+      () => {
```

Comment:
do we need this function now?

Reply:
i took reference from signup page so i left it the same

Reply:
is it the anonymous function in line 42 or the uploadTask.on function , with the uploadTask function i'm updating in the database.

Reply:
anonymous function, which is basically empty

Reply:
(snapshot) => {
        const progress = Math.round((snapshot.bytesTransferred / snapshot.totalBytes) * 100);
        setProgress(progress);
      }
I thought of adding it for progress function

Reply:
that'd require more implementation. Let's leave it for now